### PR TITLE
Custom actions

### PR DESCRIPTION
**Pull request by nskartoredjo**
As a developer I want to buy land by giving: 
- the width and depth of the desired land,
- the distance from the road,
- if I want to buy land which contains buildings or not,
- the price per squared meter,
- the zone where the desired land should be bought.

So that the GOAL agent should not worry about the multi-polygon calculations.

```
buy_specified_land(SID, ZoneID, BuildingInd, Price, Depth, Width, DistanceToRoad):

SID: stakeholder ID.
ZoneID: Zone ID.
BuildingInd: A indicator from 0 to 2 which specifies the desired land to:
- 0: exclude building areas.
- 1: only use building areas.
- 2: include building areas.
Price: the price per squared meter.
Depth: The depth of the desired land, if, in addition to the depth, it equals zero, this specification will be skipped.
Width: The width of the desired land  if, in addition to the width, it equals zero, this specification will be skipped.
DistanceToRoad: the distance from the desired land from the road. If zero, this specification will be skipped.

```

### CODE REVIEW

File: contextvh/src/main/java/contextvh/ContextEntity.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public ContextEntity(final EntityListener env, final String intendedStakeholder,
 
 	@Override
 	public Percept performAction(final Action action) throws TranslationException {
+		System.out.println(action);
```
Comment (jannelie) [inline contextvh/src/main/java/contextvh/ContextEntity.java]: Why do you have this print action?

File: contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+	private static final MapType MAPTYPE = MapType.MAQUETTE;
+
+	/**
+	 * this method includes multiple procedures:
+	 * 
+	 * - It retrieves the values.
+	 * 
+	 * - within the parameter list. If there are more parameters, then it will
+	 * return a exception.
```
Comment (HarmenKroon) [inline contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java]: an exception

File: contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+			Number width = ((Numeral) params.next()).getValue();
+			Number distanceToRoad = ((Numeral) params.next()).getValue();
+
+			if (params.hasNext())
+				throw new TranslationException("buy_specified_land: to many arguments");
+
+			MultiPolygon buyableLand = getBuyableLand(sID.intValue(), zoneID.intValue());
+
+			if (buyOnBuilding.intValue() == 0)
+				buyableLand = excludeBuildingLand(buyableLand);
```
Comment (HarmenKroon) [inline contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java:101]: 0: exclude buildings
1: include buildings or only buildings?
2: only buildings or include buildings?

**Comment by nskartoredjo** [comment]
Also, testing and checkstyle would be added by using new branches from the develop branch, after the merge.

File: contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+	 * @param depth
+	 *            the specified double depth If zero in addition to width, this
+	 *            specification will be skipped.
+	 * @param width
+	 *            the specified double width. If zero in addition to depth, this
+	 *            specification will be skipped.
+	 * @param distanceToRoad
+	 *            the distance the new area should be from a road. If zero, this
+	 *            check will be skipped.
+	 * @return
```
Comment (HarmenKroon) [inline contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java:288]: Create actionspec with explanation about zeros and skips

**Comment by HarmenKroon** [comment]
Why is travis changed?

**Comment by nskartoredjo** [comment]
@HarmenKroon it skips the checkstyle and coverage. See previous comment

**Comment by ardwennem2** [comment]
Approved

**Comment by HarmenKroon** [comment]
Approved

**Comment by jannelie** [comment]
Approved

**Comment by jannelie** [comment]
Approved once again for pullapprove 

[Approved with PullApprove](https://pullapprove.com/nskartoredjo/tygron/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=nskartoredjo/tygron)

**Comment by HarmenKroon** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/nskartoredjo/tygron/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=nskartoredjo/tygron)

**Pull request merged**